### ISSUE TITLE
symfony 4 producer is not visible in console commands

### ISSUE THREAD

**Issue by mkingbe** [php-amqplib/RabbitMqBundle#501]
Hello,

I have a console command that needs access to a producer in this way:

`$this->getContainer()->get('entity-update-producer');`

After I upgrade to symfony 4 I couldn't make the console command to work:


```
In Container.php line 252:
                                                                                                                                                                                                                           
  The "entity-update-producer" service or alias has been removed or inlined when the container was compiled. You should either make it public, or stop using the container directly and use dependency injection instead.  

```


What should I change in config or console command in order to work ?

Thanks,

**Comment by alexlongshaw** [comment]
I'm making a few assumptions here as I haven't used Symfony 4 much.

It looks like you're hitting a ServiceNotFoundException here: https://github.com/symfony/dependency-injection/blob/4.0/Container.php#L252

So with Symfony 4, it seems that services are private by default. This means that you'll have to set services as public to access them from the container http://symfony.com/doc/current/service_container/alias_private.html.

`services:
    # this makes public all the services defined in this file
    _defaults: { public: true }

    # you can also make public individual services
    App\Manager\UserManager:
        public: true`

**Comment by stloyd** [comment]
Should be fixed by #500.

**Comment by mkingbe** [comment]
I fixed this by using dependency injection instead of grabbing the service directly from container.

```
    App\Command\SimulateBrandUpdateCommand:
        class: App\Command\SimulateBrandUpdateCommand
        arguments:
            - '@entity-update-producer'
```

Is working for me.
